include/nbl/builtin/hlsl/spirv_intrinsics/core.hlsl

```diff
@@ -47,199 +69,1272 @@ static const uint32_t3 LocalInvocationId;
 static const uint32_t3 GlobalInvocationId;
 [[vk::ext_builtin_input(spv::BuiltInLocalInvocationIndex)]]
 static const uint32_t LocalInvocationIndex;
+[[vk::ext_builtin_input(spv::BuiltInSubgroupSize)]]
+static const uint32_t SubgroupSize;
+[[vk::ext_builtin_input(spv::BuiltInNumSubgroups)]]
+static const uint32_t NumSubgroups;
+[[vk::ext_builtin_input(spv::BuiltInSubgroupId)]]
+static const uint32_t SubgroupId;
+[[vk::ext_builtin_input(spv::BuiltInSubgroupLocalInvocationId)]]
+static const uint32_t SubgroupLocalInvocationId;
+[[vk::ext_builtin_input(spv::BuiltInVertexIndex)]]
+static const uint32_t VertexIndex;
+[[vk::ext_builtin_input(spv::BuiltInInstanceIndex)]]
+static const uint32_t InstanceIndex;
+[[vk::ext_builtin_input(spv::BuiltInSubgroupEqMask)]]
+static const uint32_t4 SubgroupEqMask;
+[[vk::ext_builtin_input(spv::BuiltInSubgroupGeMask)]]
+static const uint32_t4 SubgroupGeMask;
+[[vk::ext_builtin_input(spv::BuiltInSubgroupGtMask)]]
+static const uint32_t4 SubgroupGtMask;
+[[vk::ext_builtin_input(spv::BuiltInSubgroupLeMask)]]
+static const uint32_t4 SubgroupLeMask;
+[[vk::ext_builtin_input(spv::BuiltInSubgroupLtMask)]]
+static const uint32_t4 SubgroupLtMask;
+}
 
-//! General Types
-template<uint32_t StorageClass, typename T>
-using pointer_t = vk::SpirvOpaqueType<spv::OpTypePointer,vk::Literal<vk::integral_constant<uint32_t,StorageClass> >,T>;
+//! Execution Modes
+namespace execution_mode
+{
+	void invocations()
+	{
+		vk::ext_execution_mode(spv::ExecutionModeInvocations);
+	}
```

Comment:
have you tested this works with godbolt?

AFAIK this only works if you call directly from entry point

Reply:
nvm we did it this way before, all is good